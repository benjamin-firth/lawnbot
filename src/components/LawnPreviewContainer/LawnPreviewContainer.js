import React from 'react';
import { useSelector } from 'react-redux'
import styled from 'styled-components';
import LawnPreviewCard from '../LawnPreviewCard/LawnPreviewCard';
import { exampleYards } from '../../utils/exampleYards';

const LawnPreviewContainer = () => {
  const currentYardSize = useSelector(state => state.currentYardSize)

  const displayPreviewCards = () => {
    let yards;
    if (currentYardSize === null || isNaN(currentYardSize)) {
      yards = exampleYards;
    } else {
      yards = exampleYards.filter(yard => yard.yardSize < currentYardSize + 1000 && yard.yardSize > currentYardSize - 1000 );
    };
    
    const displayYards = yards.map(({ beforePic, afterPic, services, location }) => {
      return (
        <LawnPreviewCard 
          beforePic={beforePic} 
          afterPic={afterPic}
          location={location}
          services={services} 
          key={Date.now() * Math.random()}/>
      );
    });

    if (displayYards.length > 0) {
      return displayYards;
    } else {
      return (
        <StyledAlert>
          <h2>No yards match your size</h2>
        </StyledAlert>
      )
    };
  };

  return (
    <StyledPreviewSection>
      { displayPreviewCards() }
    </StyledPreviewSection>
  );
};

const StyledPreviewSection = styled.section`
  display: flex;
  flex-direction: column;
  height: 80vh;
  overflow: scroll;
  padding-bottom: 10px;
  width: 45vw;
`;

const StyledAlert = styled.div`
  align-items: center;
  color: #6a6a6a;
  display: flex;
  height: 80vh;
  justify-content: center; 
  width: 45vw;
`;

export default LawnPreviewContainer;