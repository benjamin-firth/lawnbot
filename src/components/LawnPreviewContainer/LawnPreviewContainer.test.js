  
import React from 'react';
import { render } from '@testing-library/react';
import LawnPreviewContainer from './LawnPreviewContainer';
import { shallow } from 'enzyme';

jest.mock("react-redux", () => ({
  useSelector: () => [{currentYardSize: 1000}],
}));

describe('LawnPreviewContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LawnPreviewContainer />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});