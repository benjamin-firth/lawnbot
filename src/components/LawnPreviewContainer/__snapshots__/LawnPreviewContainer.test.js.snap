// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LawnPreviewContainer should match the snapshot 1`] = `
<styled.section>
  <LawnPreviewCard
    afterPic="lawn-after-small.jpg"
    beforePic="lawn-before-small.jpg"
    key="203070453760.30392"
    services={
      Array [
        "Lawn Seeding",
        "Mowing",
        "Tick Spray",
      ]
    }
  />
  <LawnPreviewCard
    afterPic="lawn-after-small.jpg"
    beforePic="lawn-before-small.jpg"
    key="1514668527609.096"
    services={
      Array [
        "Lawn Seeding",
        "Mowing",
      ]
    }
  />
  <LawnPreviewCard
    afterPic="lawn-after-small.jpg"
    beforePic="lawn-before-small.jpg"
    key="956227266886.5725"
    services={
      Array [
        "Tick Spray",
      ]
    }
  />
  <LawnPreviewCard
    afterPic="lawn-after-small.jpg"
    beforePic="lawn-before-small.jpg"
    key="13867196503.99524"
    services={
      Array [
        "Lawn Seeding",
        "Mowing",
        "Tick Spray",
        "Edging",
      ]
    }
  />
  <LawnPreviewCard
    afterPic="lawn-after-small.jpg"
    beforePic="lawn-before-small.jpg"
    key="41058368036.561195"
    services={
      Array [
        "Mowing",
      ]
    }
  />
</styled.section>
`;
